ToDoApp/ToDoApp/SignUp/SignUpViewModel.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  SignUpViewModel.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-20.
+//
+
+import Foundation
+import FirebaseAuth
+enum SignUpErrors: Identifiable, Hashable {
+    var id: Int {
+        hashValue
+    }
+    case invalidEmail
+    case invalidPassword
+    case invalidName
+    case authenticationError
+}
+@MainActor
+class SignUpViewModel: ObservableObject {
+    // MARK: Internal Properties
+    @Published var error: SignUpErrors?
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var name: String = ""
+    @Published var isCheckBoxTapped: Bool = false
```

Comment:
sometimes names should be less general to better explain what kind of check box it relates to
```suggestion
    @Published var isAcceptTermsAndConditionsChecked: Bool = false
```

Reply:
also if there is a separate method to toggle this state, then this var should be private(set)